packages/core/src/application_init.ts

```diff
@@ -68,6 +68,8 @@ export class ApplicationInitStatus {
         const initResult = this.appInits[i]();
         if (isPromise(initResult)) {
           asyncInitPromises.push(initResult);
+        } else if (isObservable(initResult)) {
+          asyncInitPromises.push(initResult.toPromise());
```

Comment:
> It seems that lastValueFrom is not available in rxjs 6 yet, maybe we can have another PR when rxjs 7 released?

Makes sense, thanks @vthinkxie. I think we should add a quick comment above that line to provide more context for the followup refactoring (once we update RxJS to v7):

```suggestion
          // Note: `.toPromise()` will be deprecated in RxJS v7 and removed completely in later versions.
          // Once RxJS is updated in the repo to v7, this code should be refactored and the `lastValueFrom` function 
          // should be used instead.
          asyncInitPromises.push(initResult.toPromise());
```

We have [an item on the roadmap](https://angular.io/guide/roadmap#ensure-smooth-adoption-for-future-rxjs-changes-v7-and-beyond) to update RxJS to v7 and if it happens before next major release (v12), we could update that code too. Otherwise, we'll update it once RxJS is updated and the `lastValueFrom` function becomes available.

Reply:
@vthinkxie it looks like the comment was reverted, could you please add it back, so there is some context for future refactoring? Thank you.

Reply:
sure, wait one moment

> @vthinkxie it looks like the comment was reverted, could you please add it back, so there is some context for future refactoring? Thank you.

sure, wait one moment